docs/concepts/runtime_capabilities.md

```diff
@@ -103,21 +103,59 @@ Both the [`elastic` connector](../reference/connectors/elastic.md) and the [`htt
 
-Connectors like [`kafka_consumer`](../reference/connectors/kafka.md#consumer) [`wal`](../reference/connectors/wal.md) support event delivery acknowledgements and are well suited for implementing workloads that can guarantee **at-least-once* delivery.
+Connectors like [`kafka_consumer`](../reference/connectors/kafka.md#consumer) [`wal`](../reference/connectors/wal.md) support event delivery and processing acknowledgements and are well suited for implementing workloads that can guarantee *at-least-once* delivery. This guarantee can, however, only hold if all participants honor the same guarantee.
 
-For the circuit breaker mechanism a successful event acknowledgement is used a signal for the circuit breaker to close as the downstream system seems to be back to normal.
+For the circuit breaker mechanism, a successful event acknowledgment is used as a signal for the circuit breaker to close as the downstream system seems to be back to normal.
+
+Please note that contraflow events, be it circuit breaker or GD events, do not traverse over connectors. In other words, two pipelines connected by a connector will not see each other's contraflow events.
+
+### Example 1: kafka source & udp sink
+
+**Source**: The kafka source supports guaranteed delivery. A failure on a message after the latest acknowledged offset will revert to that message and replay all data since then.
+
+**Sink**: The udp sink has no delivery guarantee any message reaching it will be automatically delivered
+
+**Result**: All messages read from Kafa will be automatically acknowledged.
+
+### Example 2: udp source & kafka sink
+
+**Source**: Upd as transport has no guarantees, the upstream system sending the messages does not know if it arrives or is handled.
+
+**Sink**: Depending on its setting, Kafka acknowledges successful storage of the message and will fail an event if it can't store it.
+
+**Result**: The upstream system has no guarantees its messages reach somewhere. If a message is delivered to tremor it will make a best effort attempt to deliver it, but if Kafka fails to persist the event, a replay can't happen due to UDP's limitation.
+
+### Example 3: http source & kafka sink
+
+**Source**: The WebSocket source will acknowledge successful delivery of the message to its upstream system but does not support replays or other methods of dealing with failures.
+
+**Sink**: Depending on its setting, Kafka acknowledges successful storage of the message and will fail an event if it can't store it.
```

Comment:
```suggestion
**Sink**: Depending on its setting, `kafka_producer` acknowledges successful storage of the message and will fail an event if it can't store it.
```